### PR TITLE
Add LookBook Integration

### PR DESCRIPTION
**Pull request by cirdes**
**Overview:**
This pull request introduces the LookBook tool to our project. LookBook is an interactive UI component library explorer and playground that allows developers to preview, document, and test UI components in isolation.

Screenshot 2024-07-27 at 4 21 25 PM

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -20,7 +20,10 @@ gem "stimulus-rails", "1.3.3"
 # Bundle and process CSS [https://github.com/rails/cssbundling-rails]
 gem "cssbundling-rails", "1.4.0"
 
-# Build JSON APIs with ease [https://github.com/rails/jbuilder]
+gem "lookbook", "2.3.2"
+
+gem "lucide-rails", "0.4.0"
```
Comment (cirdes) [inline Gemfile:25]: it will make our life easier when rendering icons

File: config/initializers/lookbook.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# lookbook workaround for phlex components
+if defined?(Lookbook)
```
Comment (cirdes) [inline config/initializers/lookbook.rb:4]: Lookbook has some compatibility issues with Phlex that will be fixed in Phlex 2.0.

File: test/components/previews/phlex_ui/theme_toggle_preview.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            Button(variant: :ghost, icon: true) do
+              svg(
+                xmlns: "http://www.w3.org/2000/svg",
+                viewbox: "0 0 24 24",
+                fill: "currentColor",
+                class: "w-4 h-4"
+              ) do |s|
+                s.path(
+                  d:
+                    "M12 2.25a.75.75 0 01.75.75v2.25a.75.75 0 01-1.5 0V3a.75.75 0 01.75-.75zM7.5 12a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM18.894 6.166a.75.75 0 00-1.06-1.06l-1.591 1.59a.75.75 0 101.06 1.061l1.591-1.59zM21.75 12a.75.75 0 01-.75.75h-2.25a.75.75 0 010-1.5H21a.75.75 0 01.75.75zM17.834 18.894a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 10-1.061 1.06l1.59 1.591zM12 18a.75.75 0 01.75.75V21a.75.75 0 01-1.5 0v-2.25A.75.75 0 0112 18zM7.758 17.303a.75.75 0 00-1.061-1.06l-1.591 1.59a.75.75 0 001.06 1.061l1.591-1.59zM6 12a.75.75 0 01-.75.75H3a.75.75 0 010-1.5h2.25A.75.75 0 016 12zM6.697 7.757a.75.75 0 001.06-1.06l-1.59-1.591a.75.75 0 00-1.061 1.06l1.59 1.591z"
```
Comment (iseth) [inline test/components/previews/phlex_ui/theme_toggle_preview.rb:18]: could we use the helper here :)

File: test/components/previews/phlex_ui/theme_toggle_preview.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+              svg(
+                xmlns: "http://www.w3.org/2000/svg",
+                viewbox: "0 0 24 24",
+                fill: "currentColor",
+                class: "w-4 h-4"
+              ) do |s|
+                s.path(
+                  fill_rule: "evenodd",
+                  d:
+                    "M9.528 1.718a.75.75 0 01.162.819A8.97 8.97 0 009 6a9 9 0 009 9 8.97 8.97 0 003.463-.69.75.75 0 01.981.98 10.503 10.503 0 01-9.694 6.46c-5.799 0-10.5-4.701-10.5-10.5 0-4.368 2.667-8.112 6.46-9.694a.75.75 0 01.818.162z",
```
Comment (iseth) [inline test/components/previews/phlex_ui/theme_toggle_preview.rb:35]: same here if the helper exists

**Review by iseth — APPROVED** [review]
Looks great!

**Pull request merged**